src/Symfony/Component/HttpClient/NativeHttpClient.php

```diff
@@ -208,8 +211,7 @@ public function request(string $method, string $url, array $options = []): Respo
         $context = stream_context_create($context, ['notification' => $notification]);
         self::configureHeadersAndProxy($context, $host, $options['request_headers'], $proxy, $noProxy);
 
-        $url = implode('', $url);
-        $this->logger && $this->logger->info(sprintf('Request: %s %s', $method, $url));
+        $this->logger && $this->logger->info(sprintf('Request: %s %s', $method, $stringUrl));
 
         return new NativeResponse($this->multi, $context, $url, $options, $gzipEnabled, $info, $resolveRedirect, $onProgress, $this->logger);
```

Comment:
`s/$url/implode('', $url)` on this line then

Reply:
yes, that was one of my idea if we would have moved up this line.

Reply:
Fixed.